CI/Dockerfile

```diff
@@ -33,63 +54,207 @@ RUN apt-get -y update; \
+                                -DCMAKE_C_COMPILER=${CC} \
+                                -DBUILD_STATIC_LIBS=ON && \
+    make -j${ci_jobs} && \
+    make install && \
+    cd && \
+    rm -rf ${geant4_build_dir}
+
+
+ENV embree_install_dir=${install_dir}/embree
+ENV embree_build_dir=${build_dir}/embree
 
-ENV double_down_build_dir=${build_dir}/double-down/
-ENV double_down_install_dir=${install_dir}/double-down/
+# Clone and install Embree
+RUN mkdir -p ${embree_build_dir}/build && \
+    cd ${embree_build_dir} && \
+    git clone -b ${EMBREE_BRANCH} https://github.com/embree/embree && \
+    cd build && \
+    cmake ../embree -DCMAKE_INSTALL_PREFIX=${embree_install_dir} \
+                    -DEMBREE_TASKING_SYSTEM=INTERNAL \
+                    -DEMBREE_ISPC_SUPPORT=OFF \
+                    -DEMBREE_TUTORIALS=OFF \
+                    -DEMBREE_TBB_ROOT=/usr && \
+    make -j${ci_jobs} && \
+    make -j${ci_jobs} install && \
+    cd && \
+    rm -rf ${embree_build_dir}
 
-# Build Embree
-#TODO move sh file contents into this Dockerfile
-RUN /root/etc/CI/docker/build_embree.sh
 
 FROM external_deps AS hdf5
 
-# Set HDF5 env variable
+# accessing gloabl ARGs in build stage
+ARG HDF5_VERSION_major
+ARG HDF5_VERSION
+ARG install_dir
+ARG build_dir
+ARG ci_jobs
+ARG CXX
+ARG CC
+
 ENV hdf5_build_dir=${build_dir}/hdf5
 ENV hdf5_install_dir=${install_dir}/hdf5
 
-# Build HDF5
-# HDF5 argument possible value: 1.10.4 or system
-ARG HDF5=1.10.4
-ENV HDF5_VERSION=${HDF5}
-#TODO move sh file contents into this Dockerfile
-RUN /root/etc/CI/docker/build_hdf5.sh
+RUN mkdir -p ${hdf5_build_dir}/build && \
+    cd ${hdf5_build_dir} && \
+    wget https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-${HDF5_VERSION_major}/hdf5-${HDF5_VERSION}/src/hdf5-${HDF5_VERSION}.tar.gz && \
+    tar -xzf hdf5-${HDF5_VERSION}.tar.gz && \
+    cd build && \
+    ../hdf5-${HDF5_VERSION}/configure --enable-shared \
+        --prefix=${hdf5_install_dir} \
+        CXX=${CXX} \
+        CC=${CC} && \
+    make -j${ci_jobs} && \
+    make install && \
+    cd && \
+    rm -rf ${hdf5_build_dir}
+
 
 FROM hdf5 AS moab
 
-ENV PYTHONPATH=/root/build_dir/moab/bld/pymoab/lib/python3.8/site-packages
+# accessing gloabl ARGs in build stage
+ARG install_dir
+ARG MOAB_BRANCH
+ARG build_dir
+ARG ci_jobs
+ARG CXX
+ARG CC
+
 
 # Set MOAB env variable
 ENV moab_build_dir=${build_dir}/moab
 ENV moab_install_dir=${install_dir}/moab
 
-ARG MOAB=5.3.0
-ENV MOAB_VERSION ${MOAB}
-#TODO move sh file contents into this Dockerfile
-RUN if [ "${MOAB_VERSION}" != "master" ] && [ "${MOAB_VERSION}" != "develop" ]; then \
-        /root/etc/CI/docker/build_moab.sh; \
-    fi;
-    
\ No newline at end of file
+
+RUN mkdir -p ${moab_build_dir}/build && \
+    cd ${moab_build_dir} && \
+    git clone -b ${MOAB_BRANCH} --depth 1 https://bitbucket.org/fathomteam/moab && \
+    cd build && \
+    cmake ../moab -DCMAKE_INSTALL_RPATH=${hdf5_install_dir}/lib:${moab_install_dir}/lib \
```

Comment:
IIRC MOAB doesn't set the `CMAKE_BUILD_TYPE` by default. Adding `CMAKE_BUILD_TYPE=Release` might be worthwhile to ensure standard compiler optimization flags are included in the build.

Reply:
How about this?
```suggestion
    cmake ../moab -DCMAKE_INSTALL_RPATH=${hdf5_install_dir}/lib:${moab_install_dir}/lib \
                              -DCMAKE_BUILD_TYPE=Release \
```

Reply:
Thanks for this, that makes sense to me now. I've committed this and [now the CI is running](https://github.com/shimwell/DAGMC/actions/runs/4893025226)